asmcomp/cmmgen.ml

```diff
@@ -2881,6 +2899,13 @@ and emit_boxed_int32_constant n cont =
   else
     Csymbol_address("caml_int32_ops") :: Cint n :: cont
 
+and emit_boxed_uint32_constant n cont =
+  let n = Nativeint.of_int32 (Uint32.to_int32 n) in
+  if size_int = 8 then
+    Csymbol_address("caml_uint32_ops") :: Cint32 n :: Cint32 0n :: cont
+  else
+    Csymbol_address("caml_uint32_ops") :: Cint n :: cont
+
```

Comment:
It could make sense to factorize the two functions with a `emit_boxed_32_constant header (n : nativeint) cont` function.